test/fixtures/projects/index.js

```diff
@@ -0,0 +1,32 @@
+const fs = require('fs');
+const path = require('path');
+const android = require('./android');
+
+const dependencies = {
+  'react-native-vector-icons': {
+    'package.json': fs.readFileSync(path.join(__dirname, '../files/dependency.json')),
+    android: android.valid,
+  },
+  'rnpm-android-no-package': {
+    'package.json': fs.readFileSync(path.join(__dirname, '../files/dependency.json')),
+    android: android.noPackage,
+  },
+};
+
+const flat = {
+  android: android.valid,
+  node_modules: dependencies,
+};
+
+const nested = {
+  'android': {
+    app: android.valid,
+  },
```

Comment:
What's the difference between `android/app` and `android: {app: {}}` ? The first one just creates a folder named `android/app` - and given the nature of UNIX - it works. But for the `findAndroidProjectFolder` it's not gonna work, since the first if statement checks for `android` itself. That situation is not going to happen anytime in the future, so it's not worth reordering.